lambdas/handlers/authoriser_handler.py

```diff
@@ -34,126 +28,35 @@
 @ensure_environment_variables(names=["SSM_PARAM_JWT_TOKEN_PUBLIC_KEY"])
 @override_error_check
 def lambda_handler(event, context):
+    request_context.app_interaction = LoggingAppInteraction.LOGIN.value
+    logger.info("Authoriser handler triggered")
+    ssm_jwt_public_key_parameter = os.environ["SSM_PARAM_JWT_TOKEN_PUBLIC_KEY"]
+    auth_token = event.get("authorizationToken")
+    principal_id = ""
+    if event.get("methodArn") is None:
+        return {"Error": "methodArn is not defined"}
+    _, _, _, region, aws_account_id, api_gateway_arn = event.get("methodArn").split(":")
+    api_id, stage, _http_verb, _resource_name = api_gateway_arn.split("/")
+    path = "/" + _resource_name
+
+    policy = AuthPolicy(principal_id, aws_account_id)
+    policy.restApiId = api_id
+    policy.region = region
+    policy.stage = stage
     try:
-        request_context.app_interaction = LoggingAppInteraction.LOGIN.value
-        logger.info("Authoriser handler triggered")
-
-        ssm_service = SSMService()
-        public_key = ssm_service.get_ssm_parameter(
-            os.environ["SSM_PARAM_JWT_TOKEN_PUBLIC_KEY"], True
-        )
-
-        decoded = jwt.decode(
-            event["authorizationToken"], public_key, algorithms=["RS256"]
+        authoriser_service = AuthoriserService(path=path, _http_verb=_http_verb)
+        is_allow_policy = authoriser_service.auth_request(
+            ssm_jwt_public_key_parameter, auth_token
         )
-        request_context.authorization = decoded
-        ndr_session_id = decoded["ndr_session_id"]
-        current_session = find_login_session(ndr_session_id)
-        validate_login_session(current_session, ndr_session_id)
-
-        user_role = decoded["repository_role"]
-        principal_id = ""
-        _, _, _, region, aws_account_id, api_gateway_arn = event["methodArn"].split(":")
-        api_id, stage, _http_verb, _resource_name = api_gateway_arn.split("/")
-        path = "/" + _resource_name
-
-        policy = AuthPolicy(principal_id, aws_account_id)
-        policy.restApiId = api_id
-        policy.region = region
-        policy.stage = stage
-
-        resource_denied = validate_access_policy(_http_verb, path, user_role)
-
-        if resource_denied:
-            policy.denyMethod(_http_verb, path)
+        if is_allow_policy:
+            policy.allowMethod(_http_verb, path)
         else:
-            accepted_roles = RepositoryRole.list()
-            if user_role in accepted_roles:
-                policy.allowMethod(_http_verb, path)
-            else:
-                policy.denyMethod(_http_verb, path)
-
-        auth_response = policy.build()
-        return auth_response
+            policy.denyMethod(_http_verb, path)
 
     except AuthorisationException as e:
         logger.error(e)
         logger.error("failed to authenticate user")
```

Comment:
Is there a need for 2 logs here? Wouldn't the exception message be sufficient? Or maybe combine them as 1 log?

Reply:
Done